src/curl.cpp

```diff
@@ -1456,21 +1493,26 @@ bool S3fsCurl::SetIAMCredentials(const char* response)
     return false;
   }
 
-  if((S3fsCurl::is_ecs ? IAMCRED_KEYCOUNT_ECS : IAMCRED_KEYCOUNT) != keyval.size()){
+  if(S3fsCurl::IAM_field_count != keyval.size()){
     return false;
   }
 
-  S3fsCurl::AWSAccessKeyId       = keyval[string(IAMCRED_ACCESSKEYID)];
-  S3fsCurl::AWSSecretAccessKey   = keyval[string(IAMCRED_SECRETACCESSKEY)];
-  S3fsCurl::AWSAccessToken       = keyval[string(IAMCRED_ACCESSTOKEN)];
-  S3fsCurl::AWSAccessTokenExpire = cvtIAMExpireStringToTime(keyval[string(IAMCRED_EXPIRATION)].c_str());
+  S3fsCurl::AWSAccessToken       = keyval[string(S3fsCurl::IAM_token_field)];
+
+  if(S3fsCurl::is_ibm_auth){
+    S3fsCurl::AWSAccessTokenExpire = strtol(keyval[string(S3fsCurl::IAM_expiry_field)].c_str(), NULL, 0);
```

Comment:
Should the third argument be 10 instead of 0?  Also why call `string()` when `IAM_expiry_field` is already a string?  Repeated below.